### PR TITLE
Fix expired certificate fixture

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by wbclark — CHANGES_REQUESTED** [review]
This is failing on https://github.com/theforeman/foreman-installer/blob/develop/bin/katello-certs-check#L187-L195

I think you can work past this by regenerating it after adding `keyUsage = digitalSignature` to the extension file

**Comment by ehelms** [comment]
> This is failing on https://github.com/theforeman/foreman-installer/blob/develop/bin/katello-certs-check#L187-L195
> 
> I think you can work past this by regenerating it after adding `keyUsage = digitalSignature` to the extension file

That is already present -- https://github.com/theforeman/foreman-installer/pull/596/files#diff-a64128e78d50f7f94a452c36597cd1665f08fd7035bbbb4509c767244e74d834R2

```
Certificate:
    Data:
        Version: 3 (0x2)
        Serial Number:
            03:73:ed:79:77:a1:9b:2a:bd:e3:7d:55:55:28:5c:e0:99:eb:59:f6
        Signature Algorithm: sha256WithRSAEncryption
        Issuer: C = XX, L = Default City, O = Default Company Ltd
        Validity
            Not Before: Oct 20 02:42:49 2020 GMT
            Not After : Oct 18 02:42:49 2030 GMT
        Subject: C = XX, L = Default City, O = Default Company Ltd
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
                RSA Public-Key: (2048 bit)
                Modulus:
                    00:b6:9f:99:ac:81:4e:23:24:d2:5d:8d:8e:c0:e2:
                    17:22:87:5c:71:1c:51:7c:8e:46:69:f5:c4:ca:77:
                    7e:3b:5d:dd:85:06:f9:b2:4d:fe:e1:22:53:6a:03:
                    43:09:24:d3:66:ad:af:bc:31:e6:87:70:ea:0b:26:
                    09:30:11:33:60:e9:79:ca:62:49:aa:17:e1:de:a9:
                    02:b7:df:6a:ea:0f:26:c1:75:66:be:6d:71:60:ac:
                    34:51:ec:70:17:f9:5d:50:10:92:67:b5:e5:08:1c:
                    8b:bb:0c:39:e8:18:2c:ca:e0:7a:b6:18:94:90:92:
                    d3:ec:1e:42:ba:bf:5c:96:26:1f:c1:0f:33:24:d1:
                    ed:34:f1:7c:1e:4a:c5:1a:1b:4f:35:69:02:e7:61:
                    fe:b2:97:9b:62:83:ea:68:5e:39:9c:b6:0a:5f:c3:
                    30:aa:bc:66:1e:0b:75:84:a9:5f:82:e1:1b:6e:e1:
                    3c:1d:73:10:ff:bd:2f:5e:de:ac:63:14:8d:66:db:
                    ff:56:18:f0:ca:4a:20:99:0a:b4:25:9f:ef:53:26:
                    32:ef:11:ed:f1:6b:85:bd:b0:53:65:ac:e7:43:94:
                    87:55:dc:40:87:ad:3e:1c:1c:b2:68:92:e7:47:b8:
                    63:02:9c:fd:ff:cf:84:42:bd:b6:72:23:71:a7:c5:
                    db:49
                Exponent: 65537 (0x10001)
        X509v3 extensions:
            X509v3 Key Usage: 
                Digital Signature
    Signature Algorithm: sha256WithRSAEncryption
         19:79:e5:15:94:07:ff:bc:73:ac:15:7e:5b:29:3b:27:36:99:
         5d:97:24:de:c3:55:cd:08:4c:26:39:a8:6a:a3:ed:b6:68:a8:
         a3:29:c4:2d:db:fa:29:06:ff:5f:64:80:dc:08:7a:5e:25:da:
         2d:c5:43:78:53:b1:72:4c:f3:9f:a5:6c:0c:aa:d9:c2:2b:44:
         34:3e:4c:2f:3f:72:50:ca:5e:17:dc:23:63:94:c6:55:e7:ce:
         dd:4e:d5:a2:e4:f3:ab:76:70:8f:6d:9b:21:a9:09:fe:6b:7a:
         bb:9f:d7:58:b4:5e:8f:23:ee:aa:73:45:ec:5d:2c:46:d1:73:
         cc:cc:a2:2b:ff:af:0f:d8:29:23:12:41:82:bd:8c:77:8f:96:
         3c:b1:e0:a1:be:20:27:27:76:1e:14:75:b2:32:ce:5d:a5:13:
         c0:c9:3e:ed:57:8b:12:84:95:7b:c7:24:1d:32:0d:52:35:3c:
         b3:02:22:71:ea:94:28:ff:0a:93:61:ce:46:f6:5d:1f:be:c1:
         68:68:9b:6c:77:1a:31:91:c6:c5:67:93:ba:af:5c:91:be:62:
         50:59:2b:1a:2a:fa:17:b9:e5:d1:83:a7:4b:b6:dd:22:82:0b:
         41:32:d6:01:c7:ff:fc:b4:c6:6f:c7:a0:5d:bb:8c:ce:fd:99:
         4d:6f:25:62
```

**Comment by ehelms** [comment]
I think the problem here is, katello-certs-check claims to want to check for 'Key Encipherment' but the failure message is misleading.

https://github.com/theforeman/foreman-installer/blob/develop/bin/katello-certs-check#L188
https://github.com/theforeman/foreman-installer/blob/develop/bin/katello-certs-check#L193

File: spec/fixtures/katello-certs-check/ca.key [hunk 1]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (ekohl) [inline spec/fixtures/katello-certs-check/ca.key]: Why do we need to store the private key and CSR? They're quite useless for our purposes, right? If we do need to update, we can just generate a new one.

Comment (ehelms) [reply spec/fixtures/katello-certs-check/ca.key]: We do not. Figured they were not hurting anything to have the all the pieces present.

Comment (ekohl) [reply spec/fixtures/katello-certs-check/ca.key]: I generally don't like having unused files in a repo. I've also had bots trigger on private keys in a repo and received emails from that which me lean to not including the files. Perhaps a README on how it was generated would be nice?

**Comment by ekohl** [comment]
Btw, I think we need to pick this to all supported branches as well. Maybe more of a note to myself.

File: spec/fixtures/katello-certs-check/create_cert.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# Generate CA
+openssl genrsa -out ca.key 2048
+openssl req -x509 -new -nodes -key ca.key -sha256 -days 3650 -out cacert.crt
+
+# Generate server certificate
+openssl genrsa -out foreman.example.com.key 2048
+openssl req -new -key foreman.example.com.key -out foreman.example.com.csr
```
Comment (ekohl) [inline spec/fixtures/katello-certs-check/create_cert.sh:9]: Should the .key and .csr files be in `.gitignore`? Are there other files that should be ignored, like a serial file?

File: spec/fixtures/katello-certs-check/extensions.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[ some_ext ]
```
Comment (ekohl) [inline spec/fixtures/katello-certs-check/extensions.txt]: Is there a better name for this?

File: spec/fixtures/katello-certs-check/create_cert.sh (hunk 2 above)
Comment (ehelms) [reply spec/fixtures/katello-certs-check/create_cert.sh:9]: Done

**Review by ekohl — APPROVED** [review]

**Comment by ekohl** [comment]
@wbclark you requested changes so up to you to merge.

**Review by wbclark — APPROVED** [review]

**Pull request merged**